src/Apps/W1/Shopify/App/src/Customers/Codeunits/ShpfyCreateCustomer.Codeunit.al

```diff
@@ -172,9 +172,12 @@ codeunit 30110 "Shpfy Create Customer"
 
         Customer.Modify();
 
-        ShopifyCustomer.Copy(TempShopifyCustomer);
-        ShopifyCustomer."Customer SystemId" := Customer.SystemId;
-        ShopifyCustomer.Insert();
+        ShopifyCustomer.SetRange(Id, TempShopifyCustomer.Id);
+        if ShopifyCustomer.IsEmpty() then begin
```

Comment:
This is insufficient, on line 183 it takes `ShopifyCustomer.Id` but you don't do a get here, as such, `ShopifyCustomer.Id` will be default value. This only works if the customer doesn't exist.

You probably want `if not ShopifyCustomer.Get(TempShopifyCustomer.Id)`. I didn't check the primary key, so validate before putting this code :D